cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -38,27 +41,24 @@
 /**
  *  Handles creating indexes for a given datastore.
  */
-class IndexCreator {
+public class IndexCreator {
 
-    private final SQLDatabase database;
     private final Datastore datastore;
     private static Random indexNameRandom = new Random();
 
-    private final ExecutorService queue;
+    private final SQLDatabaseQueue queue;
 
     private static final Logger logger = Logger.getLogger(IndexCreator.class.getName());
 
-    public IndexCreator(SQLDatabase database, Datastore datastore, ExecutorService queue) {
+    public IndexCreator(Datastore datastore, SQLDatabaseQueue queue) {
```

Comment:
If you have a cast it is coupled to the implementation so you may as well be up front about it rather than allowing someone to accidentally provide some other Datastore impl that throws a CCE later. Obviously if the FTS check is moved then it doesn't apply.